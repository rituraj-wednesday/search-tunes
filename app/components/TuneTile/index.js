/**
 *
 * TuneTile
 *
 */

import React, { memo } from 'react';
import PropTypes from 'prop-types';
import styled from '@emotion/styled';
import IconButton from '@mui/material/IconButton';
import PlayCircleFilledRoundedIcon from '@mui/icons-material/PlayCircleFilledRounded';

const TrackWrapper = styled.div`
  border-radius: 4px;
  border: 3px solid #a88d8d;
  overflow: hidden;
  height: 200px;
  width: 200px;
  background-image: url(${(props) => props.trackArtURL});
  background-size: cover;
`;

/**
 * TuneTile component displays track as a tile. Also displays Preview and Info button on hover.
 *
 * @date 09/05/2024 - 16:37:00
 *
 * @param {Object} props - The component props.
 * @param {string} props.track - Track data.
 * @returns {JSX.Element} The TuneTile component.
 */
export function TuneTile({ track }) {
  return (
    <TrackWrapper trackArtURL={track.artworkUrl100} aria-label={track.trackName} data-testid="tune-tile">
      <IconButton>
        <PlayCircleFilledRoundedIcon sx={{ color: 'black' }} />
      </IconButton>
    </TrackWrapper>
  );
}

TuneTile.propTypes = {
  track: PropTypes.shape({
    artworkUrl100: PropTypes.string,
    trackName: PropTypes.string
  })
};

TuneTile.defaultProps = {
  track: {
    artworkUrl100: '',
    trackName: ''
  }
};

export default memo(TuneTile);
