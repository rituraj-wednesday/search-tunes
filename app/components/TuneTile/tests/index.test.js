/**
 *
 * Tests for TuneTile
 *
 */

import React from 'react';
// import { fireEvent } from '@testing-library/dom'
import { renderWithIntl } from '@utils/testUtils';
import TuneTile from '../index';

describe('<TuneTile />', () => {
  it('should render and match the snapshot', () => {
    const { baseElement } = renderWithIntl(<TuneTile track={{ trackName: 'hello', artworkUrl100: 'artWork' }}/>);
    expect(baseElement).toMatchSnapshot();
  });

  it('should contain 1 TuneTile component', () => {
    const dummyTrack = { trackName: 'hello', artworkUrl100: 'artWork' };
    const { getAllByLabelText } = renderWithIntl(<TuneTile  track={dummyTrack}/>);
    expect(getAllByLabelText(dummyTrack.trackName).length).toBe(1);
  });
});
