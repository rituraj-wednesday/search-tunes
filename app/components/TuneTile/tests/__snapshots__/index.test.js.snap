// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TuneTile /> should render and match the snapshot 1`] = `
<body>
  <div>
    <div
      aria-label="hello"
      class="css-1x7zvh7 enu2xcr0"
      data-testid="tune-tile"
    >
      <div
        class="css-lifrad enu2xcr1"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-wg28wd-MuiSvgIcon-root"
            data-testid="PlayCircleFilledRoundedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2m-2 13.5v-7c0-.41.47-.65.8-.4l4.67 3.5c.27.2.27.6 0 .8l-4.67 3.5c-.33.25-.8.01-.8-.4"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</body>
`;
